src/librustc/middle/resolve_lifetime.rs

```diff
@@ -89,9 +89,9 @@ struct LifetimeContext<'a, 'tcx: 'a> {
 
 #[derive(PartialEq, Debug)]
 enum ScopeChain<'a> {
-    /// EarlyScope(['a, 'b, ...], s) extends s with early-bound
-    /// lifetimes.
-    EarlyScope(&'a [hir::LifetimeDef], Scope<'a>),
+    /// EarlyScope(['a, 'b, ...], start, s) extends s with early-bound
+    /// lifetimes, with consecutive indexes from `start`.
```

Comment:
It means that `'a` has index `start`, `'b` has `start + 1`, etc. Types don't come before regions (except `Self`) in the same item, but they can be the trait/impl generics for a method's `EarlyScope`.